src/testing.erl

```diff
@@ -75,8 +75,17 @@ test_function({FileName, {M,F,A}, Type}, OrigNode, RefacNode, Options, Pid) ->
 % Gets a single function and finds the PropEr types for its arguments
 % -spec convert_type({mfa(), [type()]}) -> 
   % {atom(), atom(), [proper_types:rich_result(proper_types:fin_type())]}.
-convert_type(M, ArgTypes) ->
-    lists:map(fun(ArgType) -> get_type({M, ArgType}) end, ArgTypes).
+convert_type(FileName, M, ArgTypes) ->
+    % The proper typeserver searches for the source file in the current directory
+    case filename:basename(FileName) =:= FileName of
+        true -> lists:map(fun(ArgType) -> get_type({M, ArgType}) end, ArgTypes);
+        false -> Dir = filename:dirname(FileName),
+                 {ok, CurrDir} = file:get_cwd(),
+                 file:set_cwd(Dir),
+                 Typed = lists:map(fun(ArgType) -> get_type({M, ArgType}) end, ArgTypes),
+                 file:set_cwd(CurrDir),
+                 Typed
```

Comment:
Would a try-after make this more readable? And perhaps more robust? (Depending on your taste, the first `set_cwd` may as well go into the body of the `try`.)

```Erlang
try lists:map(fun(ArgType) -> get_type({M, ArgType}) end, ArgTypes)
after file:set_cwd(CurrDir) end
```

Reply:
Thanks for the suggestion, I modified it!